### PR TITLE
Accelerator verification

### PR DESCRIPTION
**Pull request by jlmucb**

### CODE REVIEW

File: certifier_service/certprotos/certifier.proto [hunk 1]
```diff
@@ -247,15 +253,20 @@ message proof {
   repeated proof_step steps                 = 3;
 };
 
+message accelerator_group {
+  optional string type                      = 1;
```
Comment (rgerganov) [inline certifier_service/certprotos/certifier.proto:257]: we may need to include other fields like uuid, driver version, vbios version, etc.

File: include/certifier_framework.h [hunk 2]
```diff
@@ -144,6 +144,23 @@ class domain_info {
   int    service_port_;
 };
 
+const int max_accerlerators = 4;
+class accelerator {
+public:
+  accelerator();
+  ~accelerator();
+  string accelerator_type_;
```
Comment (rgerganov) [inline include/certifier_framework.h:152]: this should be enum

File: include/certifier_framework.h [hunk 3]
```diff
@@ -144,6 +144,23 @@ class domain_info {
...
+  bool verified_;
+  string location_type_;  // in-memory, network, file
```
Comment (rgerganov) [inline include/certifier_framework.h:154]: this should be enum

File: include/certifier_framework.h [hunk 4]
```diff
@@ -144,6 +144,23 @@ class domain_info {
...
+  string file_name;
+  string network_address_;
+  byte* address_;
+  long  size_;
+  int num_certs_;
+  string* certs_;
```
Comment (rgerganov) [inline include/certifier_framework.h:160]: use `std::vector`

File: include/certifier_framework.h [hunk 5]
```diff
@@ -144,6 +144,23 @@ class domain_info {
...
 
+const int max_accerlerators = 4;
+class accelerator {
+public:
+  accelerator();
+  ~accelerator();
+  string accelerator_type_;
+  bool verified_;
+  string location_type_;  // in-memory, network, file
+  string file_name;
```
Comment (rgerganov) [inline include/certifier_framework.h:155]: what kind of `file_name` or `network_address` is associated to GPU?

File: include/certifier_framework.h [hunk 6]
```diff
@@ -163,6 +180,11 @@ class cc_trust_manager {
   string public_key_algorithm_;
   string symmetric_key_algorithm_;
 
+  int num_accelerators_;
+  accelerator  accelerators_[max_accerlerators];
```
Comment (rgerganov) [inline include/certifier_framework.h:184]: use `std::vector`

File: src/cc_helpers.cc [hunk 7]
```diff
@@ -338,6 +353,18 @@ bool certifier::framework::cc_trust_manager::initialize_oe_enclave(
 #endif
 }
 
+bool certifier::framework::cc_trust_manager::accelerator_verified(const string& acc_type) {
+  for (int i = 0; i < num_accelerators_; i++) {
+    if (acc_type == accelerators_[i].accelerator_type_)
```
Comment (rgerganov) [inline src/cc_helpers.cc:358]: what if we have several accelerators of the same type? for example several NVIDIA H100

**Comment by jlmucb** [comment]
Protobuf and code changes to support accelerators like GPUs when they are implemented

**Pull request merged**